src/components/Pagination.jsx

```diff
@@ -41,75 +42,35 @@ export const Pagination = ({
         positionOLastItemOnPage
       );
     }
-  }, [activePageIDState, itemsPerPage, setCurrentPageItems, listOfItems]);
+  }, [currentPageState, itemsPerPage, setCurrentPageItems, listOfItems]);
 
   const nextButtonDisabled = useMemo(
-    () => totalPages === activePageIDState,
-    [totalPages, activePageIDState]
+    () => totalPages === currentPageState,
+    [totalPages, currentPageState]
   );
   const prevButtonDisabled = useMemo(
-    () => activePageIDState === 1,
-    [activePageIDState]
+    () => currentPageState === 1,
+    [currentPageState]
   );
 
   function onPageChange(pageNumber) {
     if (!pageNumber) {
       throw new Error('Missing number of page!');
     }
-    setActivePageIDState(pageNumber);
+    setCurrentPageState(pageNumber);
   }
 
   function onPrevArrowChange() {
-    setActivePageIDState(activePageIDState - 1);
+    setCurrentPageState(currentPageState - 1);
```

Comment:
It's not a reliable way to update the state. Setter functions from `useState` hook are asynchronous, so you might find yourself updating values based on data that is outdated and de-synchronize your application state.

For cases like this, use a callback function that takes previous state as a param:

```js
setCurrentPageState((previousCurrentPage) => previousCurrentPage - 1);
```

which btw is an equivalent to:

```js
setCurrentPageState(function (previousCurrentPage) {
    return previousCurrentPage - 1;
});
```